backend/DjangoServer/autoshop/views.py

```diff
@@ -1,39 +1,129 @@
 from django.shortcuts import render
-from django.http import HttpResponse, HttpRequest
+from django.http import HttpResponseBadRequest, HttpRequest
 from .models import User, Vehicle
 from django.shortcuts import get_object_or_404
 from django.http import JsonResponse
 from django.contrib.auth import authenticate
 from django.views.decorators.csrf import csrf_exempt
 from django.contrib.auth.models import User as Person
 
-def userRouter(request: HttpRequest):
+@csrf_exempt
+def userRouter(request: HttpRequest, id):
     if request.method == "PUT":
-        return updateUserInfo(request)
+        return updateUserInfo(request, id)
     elif request.method == "GET":
-        return getUserInfo(request)
+        return getUserInfo(request, id)
+    elif request.method == "DELETE":
+        return deleteUserInfo(request, id)
 
-def vehicleRouter(request:HttpRequest):
+@csrf_exempt
+def vehicleRouter(request: HttpRequest, id):
     if request.method == "PUT":
-        return updateVehicleInfo(request)
+        return updateVehicleInfo(request, id)
     elif request.method == "GET":
-        return getVehicleInfo(request)
+        return getVehicleInfo(request, id)
+    elif request.method == "DELETE":
+        return deleteVehicleInfo(request, id)
+
+
+def updateVehicleInfo(request: HttpRequest, id):
+    vehicle = get_object_or_404(Vehicle, pk=id)
+    response = {}
+    myData = request.PUT
+
```

Comment:
I don't think you need to have a separate response object. Instead of adding all of these "response['name'] values, can you just return the json.dumps(vehicle) object?